scylla/src/transport/topology.rs

```diff
@@ -301,12 +400,276 @@ async fn query_keyspaces(conn: &Connection) -> Result<HashMap<String, Keyspace>,
+
+    for ((keyspace_name, table_name), (columns, partition_key_columns, clustering_key_columns)) in
+        tables_schema
+    {
+        let mut partition_key = vec!["".to_string(); partition_key_columns.len()];
+        for (position, column_name) in partition_key_columns {
+            partition_key[position as usize] = column_name;
+        }
+
+        let mut clustering_key = vec!["".to_string(); clustering_key_columns.len()];
+        for (position, column_name) in clustering_key_columns {
+            clustering_key[position as usize] = column_name;
+        }
+
+        result.insert(
+            (keyspace_name.clone(), table_name.clone()),
+            Table {
+                columns,
+                partition_key,
+                clustering_key,
+                partitioner: partitioners
+                    .remove(&(keyspace_name, table_name))
+                    .unwrap_or_default(),
+            },
+        );
+    }
+
+    Ok(result)
+}
+
+fn map_string_to_cql_type(type_: &str) -> Result<CqlType, InvalidCqlType> {
+    use CollectionType::*;
+
+    let frozen = type_.starts_with("frozen<");
+
+    let type_ = if frozen {
+        let langle_position = type_.find('<').unwrap();
+        &type_[langle_position + 1..type_.len() - 1]
+    } else {
+        type_
+    };
+
+    if type_.ends_with('>') && type_.contains('<') {
+        let langle_position = type_.find('<').unwrap();
+        let inner_type = &type_[langle_position + 1..type_.len() - 1];
+        let outer_type = &type_[..langle_position];
+        let type_ = match outer_type {
+            "map" | "tuple" => {
+                let types = inner_type.split(", ").collect::<Vec<_>>();
+                if outer_type == "map" {
+                    if types.len() != 2 {
+                        return Err(InvalidCqlType(type_.to_string()));
+                    }
+                    Map(
+                        Box::new(map_string_to_cql_type(types[0])?),
+                        Box::new(map_string_to_cql_type(types[1])?),
+                    )
+                } else {
+                    return Ok(CqlType::Tuple(
+                        types
+                            .iter()
+                            .map(|type_| map_string_to_cql_type(type_))
+                            .collect::<Result<Vec<_>, _>>()?,
+                    ));
+                }
+            }
+            "set" => Set(Box::new(map_string_to_cql_type(inner_type)?)),
+            "list" => List(Box::new(map_string_to_cql_type(inner_type)?)),
+            _ => {
+                return Err(InvalidCqlType(type_.to_string()));
+            }
+        };
+        return Ok(CqlType::Collection { frozen, type_ });
+    }
+
+    Ok(match NativeType::from_str(type_) {
+        Ok(type_) => CqlType::Native(type_),
+        _ => CqlType::UserDefinedType {
+            frozen,
+            name: type_.to_string(),
+        },
+    })
+}
+
+async fn query_table_partitioners(
```

Comment:
The CI check which runs the tests with a Cassandra instance fails because the `scylla_tables` table does not exist. This function already used to handle such a situation in one of the previous versions of this PR.